tensorflow/python/ops/nn_grad.py

```diff
@@ -203,6 +203,48 @@ def _BiasAddGrad(op, received_grad):
   return (received_grad, gen_nn_ops.bias_add_grad(out_backprop=received_grad,
                                                   data_format=data_format))
 
+@ops.RegisterGradient("BiasAddGrad")
+def _BiasAddGradGrad(op, received_grad):
+  """Gradient for the BiasAddGrad op.
+
+  Args:
+    op: BiasAddGrad op for which we are calculating gradients.
+    received_grad: The gradients passed to the BiasAddGrad op.
+    
+  Returns:
+    A single gradient Tensor for the input to BiasAddGrad (which
+    is the gradient of the bias term in BiasAdd)
+  """
+  
+  try:
+    data_format = op.get_attr("data_format")
+  except ValueError:
+    data_format = None
+    
+  #zeros = array_ops.zeros_like(op.inputs[0])
+  #return gen_nn_ops._bias_add(zeros, received_grad, data_format=data_format)
+  
+  shape = array_ops.shape(op.inputs[0])
+  rank = array_ops.rank(op.inputs[0])
+  bias_shape = array_ops.shape(received_grad)
```

Comment:
If the shape is not fully defined at this point, this adds an unnecessary dependency at runtime. So it is better to do this when the shape is known, and do your previous bias_add when it is not. For example:

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/array_ops.py#L154

If possible, please also include a benchmark result to show this is not actually slower for non-trivially large shape. tf.tile may have a few performance problems on GPU, and it is on our roadmap for improvement.